subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/artifacttransforms/spec/ArtifactTransformReportSpec.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.diagnostics.internal.artifacttransforms.spec;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+/**
+ * Contains options describing how to run an {@link org.gradle.api.tasks.diagnostics.ArtifactTransformsReportTask}, which describes what features of the data model for
+ * a project should be rendered in the output.
+ */
+public final class ArtifactTransformReportSpec {
+    @Nullable
+    private final String searchTarget;
+
+    public ArtifactTransformReportSpec(@Nullable String searchTarget) {
+        this.searchTarget = searchTarget;
+    }
+
+    public Optional<String> getSearchTarget() {
```

Comment:
Looking at this class, I can't really tell what a "search target" is. 

It looks like we only set this to the type of the transform. Why is this not called `getTransformType` or `getTypeToSearchFor` or something? 

Do you plan to make a "search target" more generic at some point as some opaque search term to use when filtering transforms to report?

I think a rename or some javadoc would be helpful here